### PR TITLE
[SPARK-20401][DOC]In the spark official configuration document, the 'spark.driver.supervise' configuration parameter specification and default values are necessary.

### PR DESCRIPTION
**Pull request by guoxiaolongzte**
## What changes were proposed in this pull request?
Use the REST interface submits the spark job.
e.g.
curl -X  POST http://10.43.183.120:6066/v1/submissions/create --header "Content-Type:application/json;charset=UTF-8" --data'{
    "action": "CreateSubmissionRequest", 
    "appArgs": [
        "myAppArgument"
    ], 
    "appResource": "/home/mr/gxl/test.jar",  
    "clientSparkVersion": "2.2.0", 
    "environmentVariables": {
        "SPARK_ENV_LOADED": "1"
    }, 
    "mainClass": "cn.zte.HdfsTest", 
    "sparkProperties": {
        "spark.jars": "/home/mr/gxl/test.jar",  
        **"spark.driver.supervise": "true",** 
        "spark.app.name": "HdfsTest", 
        "spark.eventLog.enabled": "false", 
        "spark.submit.deployMode": "cluster", 
        "spark.master": "spark://10.43.183.120:6066"
    }
}'

**I hope that make sure that the driver is automatically restarted if it fails with non-zero exit code.
But I can not find the 'spark.driver.supervise' configuration parameter specification and default values from the spark official document.**
## How was this patch tested?

manual tests

Please review http://spark.apache.org/contributing.html before opening a pull request.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
Not a big deal. Waht do you think about wrapping the command you ran with ````` ``` ... ``` ````` in the PR description?

```
curl -X POST http://10.43.183.120:6066/v1/submissions/create --header "Content-Type:application/json;charset=UTF-8" --data '{
  ...
}
```

I think this makes the PR description more readable.

**Review by srowen — APPROVED** [review]

**Comment by guoxiaolongzte** [comment]
@HyukjinKwon 
But I wrote, no use ...
My computer is window7 system.
The browser is Chrome.

**Comment by HyukjinKwon** [comment]
I will probably test it by my self latter. Thanks for taking my comment into account.

**Comment by guoxiaolongzte** [comment]
Please cut a picture, I see how you show on your computer?

**Comment by HyukjinKwon** [comment]
Yup, I will let you know when I made it. I think this does not block this PR BTW :).

**Comment by HyukjinKwon** [comment]
Yup, I will let you know when I make it. I think this does not block this PR BTW :).

File: docs/configuration.md [hunk 1]
```diff
@@ -213,6 +213,15 @@ of the most common options to set are:
     and typically can have up to 50 characters.
   </td>
 </tr>
+<tr>
+  <td><code>spark.driver.supervise</code></td>
+  <td>false</td>
+  <td>
+    If value set true, restarts the driver on failure, make sure that
```
Comment (srowen) [inline docs/configuration.md]: This sentence needs to be simplified. "If true, restarts the driver automatically if it fails with a non-zero exit status."

**Comment by HyukjinKwon** [comment]
@guoxiaolongzte, I just tested the backquotes on Windows 7 with Chrome as below:





It is a backquote `` ` `` not a single-quote `'`. Could you try again? I see you made many PRs and I just would like them to be more readable.

File: docs/configuration.md [hunk 2]
```diff
@@ -213,6 +213,15 @@ of the most common options to set are:
...
+    the driver is automatically restarted if it fails with non-zero exit code.
+    Only in Spark standalone or Mesos with cluster deploy mode.
```
Comment (HyukjinKwon) [inline docs/configuration.md]: I am sorry for asking such a question @srowen. Is it the right sentence in English?

Comment (srowen) [inline docs/configuration.md]: It'd be better to say "Only has effect in Spark standalone mode or Mesos cluster deploy mode."

Comment (guoxiaolongzte) [inline docs/configuration.md]: OK,I will fix it.

**Comment by guoxiaolongzte** [comment]
@HyukjinKwon
I will try to improve my ability to translate, sorry.

**Comment by srowen** [comment]
Merged to master/2.2

**Pull request closed without merging**